spec/lib/page_modification_tracker_spec.rb

```diff
@@ -0,0 +1,146 @@
+require "rails_helper"
+require "page_modification_tracker"
+
+RSpec.describe PageModificationTracker do
+  let(:host) { "test.host" }
+  let(:selector) { "body" }
+  let(:headers) { { "Host" => host } }
+  let(:tracker) { described_class.new(host: host, selector: selector) }
+  let(:path) { "/test-page" }
+  let(:content) { "<html><head></head><body><h1>Test Content</h1></body></html>" }
+  let(:content_hash) { Digest::SHA1.hexdigest("<body><h1>Test Content</h1></body>") }
+  let(:mock_response) do
+    instance_double(ActionDispatch::Response,
+                    body: content,
+                    headers: {})
+  end
+
+  before do
+    allow_any_instance_of(ActionDispatch::Integration::Session)
+      .to receive(:get)
+      .with(path, headers: headers)
+      .and_return(mock_response)
+
+    allow_any_instance_of(ActionDispatch::Integration::Session)
+      .to receive(:response)
+      .and_return(mock_response)
+
+    allow_any_instance_of(GetIntoTeachingApiClient::TeachingEventsApi)
+      .to receive(:search_teaching_events)
+      .and_return([])
+
+    allow(::Pages::Frontmatter)
+      .to receive(:list)
+      .and_return({ path => { draft: false } })
+  end
+
+  describe "#track_page_modifications" do
+    subject { tracker.track_page_modifications }
+
+    context "when content has not changed" do
+      before do
+        PageModification.create!(path: path, content_hash: content_hash)
+      end
+
+      it "does not update the page modification" do
+        expect { subject }.not_to(
+          change { PageModification.find_by(path: path).updated_at },
+        )
+      end
+    end
+
```

Comment:
could you add a test to prove that a change to the `authenticity_token` does not invalidate the hash